Generic driver sample fails build for MikroElektronika H Bridge 4 Click shield

### Describe the bug

Building the stepper driver example with the mikroe_h_bridge_4_click shield fails to build due to the shield not having been updated to the new stepper API. This should probably have been part of #91979.

### Regression

- [x] This is a regression.

### Steps to reproduce

Steps to reproduce:
1. `west build --pristine --board mikroe_clicker_ra4m1  --shield mikroe_h_bridge_4_click samples/drivers/stepper/generic`

### Relevant log output

```shell
$ west build --pristine --board mikroe_clicker_ra4m1  --shield mikroe_h_bridge_4_click samples/drivers/stepper/generic
-- west build: making build dir <censored>/zephyrproject/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: <censored>/zephyrproject/zephyr/samples/drivers/stepper/generic
-- CMake version: 3.31.6
-- Found Python3: <censored>/tmp/venv<censored>/zephyrproject/bin/python3.13 (found suitable version "3.13.5", minimum required is "3.12") found components: Interpreter
-- Cache files will be written to: <censored>/.cache/zephyr
-- Zephyr version: 4.4.99 (<censored>/zephyrproject/zephyr)
-- Found west (found suitable version "1.5.0", minimum required is "0.14.0")
-- Board: mikroe_clicker_ra4m1, qualifiers: r7fa4m1ab3cfm
-- Shield(s): mikroe_h_bridge_4_click
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 1.0.1 (<censored>/opt/zephyr-sdk-1.0.1)
-- Found toolchain: zephyr 1.0.1 (<censored>/opt/zephyr-sdk-1.0.1)
-- Found Dtc: <censored>/opt/zephyr-sdk-1.0.1/hosttools/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.7.0", minimum required is "1.4.6")
-- Found BOARD.dts: <censored>/zephyrproject/zephyr/boards/mikroe/clicker_ra4m1/mikroe_clicker_ra4m1.dts
-- Found devicetree overlay: <censored>/zephyrproject/zephyr/boards/shields/mikroe_h_bridge_4_click/mikroe_h_bridge_4_click.overlay
devicetree error: 'en-gpios' appears in /h_bridge_4_click in <censored>/zephyrproject/zephyr/build/zephyr/zephyr.dts.pre, but is not declared in 'properties:' in <censored>/zephyrproject/zephyr/dts/bindings/stepper/zephyr,h-bridge-stepper-ctrl.yaml
CMake Error at <censored>/zephyrproject/zephyr/cmake/modules/dts.cmake:326 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  <censored>/zephyrproject/zephyr/cmake/modules/dts.cmake:447 (dts_edt_pickle)
  <censored>/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:132 (dts_init)
  <censored>/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:132 (cmake_language)
  <censored>/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  <censored>/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:6 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=<censored>/tmp/venv<censored>/zephyrproject/bin/python3.13 -DWEST_TOPDIR=<censored>/zephyrproject -DWEST_VERSION=1.5.0 -B<censored>/zephyrproject/zephyr/build -GNinja '-DWEST_PYTHON_PROPERTIES="Python;3;13;5;64;;cpython-313-x86_64-linux-gnu.so;abi3;/usr/lib/python3.13;<censored>/tmp/venv<censored>/zephyrproject/lib/python3.13;<censored>/tmp/venv<censored>/zephyrproject/lib/python3.13/site-packages;<censored>/tmp/venv<censored>/zephyrproject/lib/python3.13/site-packages"' -DBOARD=mikroe_clicker_ra4m1 -DSHIELD=mikroe_h_bridge_4_click -S<censored>/zephyrproject/zephyr/samples/drivers/stepper/generic
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- Commit: current main (aa50e830c2e3fc21cf9b739a900b06784c6c86c5)

### Additional Context

- Last good commit: c4a4d8aa873bb5d49366eb0019e51f824a95e15e
- Introduced by commit: 83956260f77e6138ca56614e004ee891d27e80ff

--- a/boards/shields/mikroe_h_bridge_4_click/mikroe_h_bridge_4_click.overlay
+++ b/boards/shields/mikroe_h_bridge_4_click/mikroe_h_bridge_4_click.overlay
@@ -5,7 +5,7 @@
 
 / {
 	aliases {
-		stepper = &mikroe_h_bridge_4_click;
+		stepper-ctrl = &mikroe_h_bridge_4_click;
 	};
 };
 
@@ -14,7 +14,6 @@
 		status = "okay";
 		compatible = "zephyr,h-bridge-stepper-ctrl";
 
-		en-gpios = <&mikrobus_header 2 GPIO_ACTIVE_LOW>;
 		gpios = <&mikrobus_header 0 GPIO_ACTIVE_LOW>, /* IN1 */
 			<&mikrobus_header 1 GPIO_ACTIVE_LOW>, /* IN2 */
 			<&mikrobus_header 6 GPIO_ACTIVE_LOW>, /* IN3 */

--- a/samples/drivers/stepper/generic/src/main.c
+++ b/samples/drivers/stepper/generic/src/main.c
@@ -13,7 +13,18 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(stepper_generic, CONFIG_STEPPER_LOG_LEVEL);
 
+/*
+ * The stepper-driver alias is optional: some stepper controllers (e.g. h-bridge based ones)
+ * drive the motor directly and do not have a separate hardware driver to enable/disable.
+ */
+#if DT_NODE_EXISTS(DT_ALIAS(stepper_driver))
+#define HAS_STEPPER_DRIVER 1
+#define STEPPER_DRIVER_MICRO_STEP_RES DT_PROP_OR(DT_ALIAS(stepper_driver), micro_step_res, 1)
 static const struct device *stepper_driver = DEVICE_DT_GET(DT_ALIAS(stepper_driver));
+#else
+#define HAS_STEPPER_DRIVER 0
+#define STEPPER_DRIVER_MICRO_STEP_RES 1
+#endif
 static const struct device *stepper_ctrl = DEVICE_DT_GET(DT_ALIAS(stepper_ctrl));
 
 enum stepper_mode {
@@ -29,7 +40,7 @@ enum stepper_mode {
 static atomic_t stepper_mode = ATOMIC_INIT(STEPPER_MODE_DISABLE);
 
 static int32_t ping_pong_target_position = CONFIG_STEPS_PER_REV * CONFIG_PING_PONG_N_REV *
-					   DT_PROP_OR(DT_ALIAS(stepper_driver), micro_step_res, 1);
+					   STEPPER_DRIVER_MICRO_STEP_RES;
 
 static K_SEM_DEFINE(stepper_generic_sem, 0, 1);
 
@@ -71,6 +82,12 @@ int main(void)
 		LOG_ERR("Device %s is not ready", stepper_ctrl->name);
 		return -ENODEV;
 	}
+#if HAS_STEPPER_DRIVER
+	if (!device_is_ready(stepper_driver)) {
+		LOG_ERR("Device %s is not ready", stepper_driver->name);
+		return -ENODEV;
+	}
+#endif
 	LOG_DBG("stepper is %p, name is %s", stepper_ctrl, stepper_ctrl->name);
 
 	stepper_ctrl_set_event_cb(stepper_ctrl, stepper_callback, NULL);
@@ -81,7 +98,9 @@ int main(void)
 		k_sem_take(&stepper_generic_sem, K_FOREVER);
 		switch (atomic_get(&stepper_mode)) {
 		case STEPPER_MODE_ENABLE:
+#if HAS_STEPPER_DRIVER
 			stepper_enable(stepper_driver);
+#endif
 			LOG_INF("mode: enable");
 			break;
 		case STEPPER_MODE_PING_PONG_RELATIVE:
@@ -107,7 +126,9 @@ int main(void)
 			LOG_INF("mode: stop");
 			break;
 		case STEPPER_MODE_DISABLE:
+#if HAS_STEPPER_DRIVER
 			stepper_disable(stepper_driver);
+#endif
 			LOG_INF("mode: disable");
 			break;
 		default:

--- a/samples/drivers/stepper/generic/tests.yaml
+++ b/samples/drivers/stepper/generic/tests.yaml
@@ -6,3 +6,11 @@ tests:
     tags: stepper
     platform_allow: nucleo_g071rb
     depends_on: gpio input
+  sample.drivers.stepper.generic.h_bridge_4_click:
+    build_only: true
+    tags:
+      - stepper
+      - shield
+    platform_allow: mikroe_clicker_ra4m1
+    extra_args: SHIELD=mikroe_h_bridge_4_click
+    depends_on: gpio

